libs/testserver/server.go

```diff
@@ -37,17 +42,36 @@ func New(t testutil.TestingT) *Server {
 	t.Cleanup(server.Close)
 
 	return &Server{
-		Server: server,
-		Mux:    mux,
-		t:      t,
+		Server:         server,
+		Mux:            mux,
+		t:              t,
+		mu:             &sync.Mutex{},
+		fakeWorkspaces: map[string]*FakeWorkspace{},
 	}
 }
 
-type HandlerFunc func(req *http.Request) (resp any, statusCode int)
+type HandlerFunc func(fakeWorkspace *FakeWorkspace, req *http.Request) (resp any, statusCode int)
 
 func (s *Server) Handle(pattern string, handler HandlerFunc) {
 	s.Mux.HandleFunc(pattern, func(w http.ResponseWriter, r *http.Request) {
-		resp, statusCode := handler(r)
+		// For simplicity we process requests sequentially. It's fast enough because
```

Comment:
Could you include results of `hyperfine -w 2 'go test ./acceptance -count=1'` before and after this change to quantify "fast enough"?

Reply:
Looks like 7% difference:

```
After

Benchmark 1: go test ./acceptance -count=1
  Time (mean ± σ):      6.268 s ±  0.312 s    [User: 8.019 s, System: 7.185 s]
  Range (min … max):    5.872 s …  6.945 s    10 runs

Before

Benchmark 1: go test ./acceptance -count=1
  Time (mean ± σ):      5.949 s ±  0.136 s    [User: 7.483 s, System: 7.051 s]
  Range (min … max):    5.777 s …  6.225 s    10 runs
```

If we narrow the critical section, we can get an extra ~100ms, but I don't think it's worth it.